### PR TITLE
Full test coverage

### PR DESCRIPTION
**Pull request by rekmarks**
- Add tests to achieve 100% test coverage for all measures (line, branch etc.)
- Require 100% test coverage
- Check coverage in CI

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1921,16 +1921,6 @@ set-blocking@^2.0.0:
   resolved "https://registry.yarnpkg.com/set-blocking/-/set-blocking-2.0.0.tgz#045f9782d011ae9a6803ddd382b24392b3d890f7"
   integrity sha1-BF+XgtARrppoA93TgrJDkrPYkPc=
 
-set-value@^2.0.0, set-value@^2.0.1:
```
Comment (Gudahtt) [inline yarn.lock]: Huh. Why was this removed?

Comment (Gudahtt) [reply yarn.lock]: Ah, it was out of date on `master`. Addressed in #60

Comment (rekmarks) [reply yarn.lock]: I think I erroneously included them during rebasing/merging. Running `yarn why set-value` on `master` yields no results.

Comment (rekmarks) [reply yarn.lock]: Fixed by #60

File: test/engine.spec.js [hunk 2]
```diff
@@ -286,4 +354,82 @@ describe('basic tests', function () {
...
+  })
+
+  it('handles batch request processing error', function (done) {
+    const origPromiseAll = Promise.all
+    Promise.all = () => {
+      Promise.all = origPromiseAll
+      throw new Error('foo')
+    }
+
+    const engine = new RpcEngine()
```
Comment (Gudahtt) [inline test/engine.spec.js]: Woah, mutating `Promise`? That seems a bit excessive. That could lead to some pretty nasty test failures if it was left unrestored somehow. I think you could stub out `_promiseHandle` here instead, so that only the instance under test is mutated.

Maybe something like this (though I didn't actually test this):
```suggestion
    const engine = new RpcEngine()
    sinon.stub(engine, '_promiseHandle').throws('foo')
```

The same goes for the following test as well

Comment (rekmarks) [reply test/engine.spec.js]: Look man, we're all professionals here 😅 

62d731d

Comment (rekmarks) [reply test/engine.spec.js]: That's: 62d731d

**Review by Gudahtt — APPROVED** [review]
Looks great! Tests are simple and easy to read
